### DISCUSSION TITLE
Merging the project into firefox (bugzilla)

### DISCUSSION THREAD

**Discussion by pachainti** [filips123/PWAsForFirefox#19] [General]
Hi,
I filed a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1719080) on bugzilla about "merging" this project into firefox. Unfortunately, I do not have the necessary skills to participate as developer. Maybe, I can help with testing.
Thank you

**Comment by filips123** [comment]
I moved the issue to a discussion as that is more appropriate.

Unfortunately merging this into Firefox would require quite a lot of time and rewrite of the project, so I don't think it will happen soon and I probably won't be able to do it alone. However, If anyone has enough knowledge and time to do it, I would be happy to see that and could maybe help to do it.

**Comment by markcellus** [comment]
The real question is why isnt this already inside of the official Firefox app? PWAs are supported in almost every other modern browser.

**Reply by black7375** [reply]
Mozilla's explanation
- https://9to5google.com/2021/01/27/firefox-discontinues-work-pwa-desktop/
- https://www.i-programmer.info/news/87-web-development/14261-firefox-drops-support-for-pwa.html

TBH, I don't know why they did it.

**Comment by black7375** [comment]
> There isn't a good place. Our current position is that we don't want to pursue this feature, see bug 1602117 etc.

It was decided as `won't fix`.

**Reply by pachainti** [reply]
In my humble opinion, this is a huge mistake. First, mozilla pioneered web app with firefox OS which would have allowed to install any app without any store, it was a step ahead compared to the other OS. Unfortunately, firefox OS died (no, it is [still alive](https://www.kaiostech.com/kaios-technologies-and-mozilla-partner-to-enable-a-healthy-mobile-internet-for-everyone/)).
PWA are the "successor" of web app. Why does mozilla not want to support it on desktop? On mobile there is a partial support. Finally, firefox is the only browser without PWA support (chromium and safari [support them](https://en.wikipedia.org/wiki/Progressive_web_application)).
I hope to be wrong, but this could be the begin of the end...

**Comment by pachainti** [comment]
Please vote and spread this https://mozilla.crowdicity.com/post/721028
